sierra_adapter/sierra_item_merger/src/main/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerWorkerService.scala

```diff
@@ -21,9 +26,24 @@ class SierraItemMergerWorkerService @Inject()(
 ) extends SQSWorker(reader, system, metrics)
     with Logging {
 
+  implicit val decodeInstant: Decoder[Instant] = new Decoder[Instant] {
+    final def apply(c: HCursor): Decoder.Result[Instant] =
+      for {
+        foo <- c.as[Int]
+      } yield {
+        Instant.ofEpochSecond(foo)
+      }
+  }
+
+  implicit val customConfig: Configuration =
+    Configuration.default.withDefaults.withDiscriminator("type")
+
   override def processMessage(message: SQSMessage): Future[Unit] =
-    JsonUtil.fromJson[SierraItemRecord](message.body) match {
-      case Success(record) => sierraBibMergerUpdaterService.update(record)
-      case Failure(e) => Future(logger.warn(s"Failed processing $message", e))
+    decode[SierraItemRecord](message.body) match {
+      case Right(record) => sierraBibMergerUpdaterService.update(record)
+      case Left(e) => Future{
+        logger.warn(s"Failed processing $message", e)
+        throw e
```

Comment:
This will return a failed future will it not? The behaviour is different from using JsonUtil as before I think.

Reply:
Yes, it is, but I think the new behaviour is the correct one? I guess we don't want the message deleted from the queue if parsing it fails, do we? We should make sure we wrap it in a SqsReaderGracefulException (or whatever) though, so that we don't fail the TryBackoff